File: week3-hash-sort/problem04-문자열내림차순으로배치하기/heetae.java
// 문제: 문자열 내림차순으로 배치하기
// 접근 방식: 정렬한다음 그걸 뒤집으면 내림차순
// 시간복잡도: O(n log n) - 자바 sort의 시간복잡도

import java.util.*;

class Solution {
    public String solution(String s) {
        char[] answer = s.toCharArray();
        Arrays.sort(answer);

        StringBuilder sb = new StringBuilder(new String(answer));
        return sb.reverse().toString();
    }
}

File: week1-array-string/problem06-문자열다루기기본/heetae.py
# 문제: 문자열 다루기 기본
# 접근 방식: 문자열의 길이가 4 또는 6인지 확인하고, 모든 문자가 숫자 범위인지 검사한다.
# 시간복잡도: O(n) (n은 s의 길이)

def solution(s):
    if len(s) not in (4, 6):
        return False
    else:
        for i in range(0, len(s)):
            if s[i] < '0' or s[i] > '9':
                return False
    return True
